### PR TITLE
Make nick completion case-insensitive

### PR DESCRIPTION
**Pull request by progval**
Closes GH-305.

### CODE REVIEW

**Review by SilverRainZ — COMMENTED** [review]
Thank you for your great work :D

File: src/sui/sui_completion.c [hunk 1]
```diff
@@ -234,7 +235,15 @@ void sui_completion_complete(SuiCompletion *self, SuiCompletionFunc *func,
     DBG_FR("Completion: prefix: '%s', suffix: '%s'",
             self->last_prefix, self->last_suffix);
 
+
+    // Go backward, to remove the prefix
+    assert(gtk_text_iter_backward_chars(&comp, g_utf8_strlen(self->last_prefix, -1)));
```
[the remark this answers is not in the document]
Comment (progval) [reply src/sui/sui_completion.c:240]: No reason in particular, I just don't like to ignore error return values. I can remove it if you prefer

[the remark this answers is not in the document]
Comment (SilverRainZ) [reply src/sui/sui_completion.c:240]: For glib application, we can use `g_warn_if_fail` (just warning)or `g_return_if_fail`(warn and return).

[the remark this answers is not in the document]
Comment (progval) [reply src/sui/sui_completion.c]: Sounds good, done!

**Pull request merged**